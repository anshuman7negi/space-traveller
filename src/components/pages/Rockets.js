import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { reserveRocket, rocketData } from '../../redux/rocket/rocketSlice';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { reserveRocket, rocketData } from '../../redux/rocket/rocketSlice';
import '../../style/Rocket.css';

const Rockets = () => {
  const dispatch = useDispatch();
  const { rockets, isLoading, isError } = useSelector((state) => state.rockets);

  useEffect(() => {
    dispatch(rocketData());
  }, [dispatch]);

  let content;

  if (isLoading) {
    content = <p>Loading...</p>;
  } else if (isError) {
    content = <p>Error fetching rocket data.</p>;
  } else {
    content = (
      <div className="RocketList">
        {rockets.map((rocket) => (
          <div key={rocket.id} className="RocketItem">
            <img src={rocket.flickr_images[0]} alt={rocket.name} />
            <div className="RocketDetails">
              <h3>{rocket.name}</h3>
              <p>{rocket.description}</p>
              <button type="button" onClick={() => dispatch(reserveRocket(rocket.id))}>Reserve Rocket</button>
            </div>
          </div>
        ))}
      </div>
    );
  }

  return <div className="RocketContainer">{content}</div>;
};

export default Rockets;
