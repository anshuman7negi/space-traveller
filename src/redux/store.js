import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './missions/missionSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
  },
});

export default store;
