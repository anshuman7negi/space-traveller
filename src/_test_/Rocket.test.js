import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store'; // Adjust the import path as needed
import Rockets from '../components/pages/Rockets';

test('Rockets component matches snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  );

  expect(asFragment()).toMatchSnapshot();
});
