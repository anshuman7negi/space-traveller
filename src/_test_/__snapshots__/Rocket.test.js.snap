// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="RocketContainer"
  >
    <p>
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
